sbtc-core/src/signer/coordinator/mod.rs

```diff
@@ -3,63 +3,18 @@ pub mod fire;
 /// Module for RObust Asynchronous Schnorr Threshold signature generation
 pub mod roast;
 
-use std::collections::HashMap;
-
-use crate::SBTCResult;
-use bitcoin::{util::taproot::TaprootSpendInfo, PublicKey, Transaction as BitcoinTransaction};
-use p256k1::ecdsa;
+use crate::{signer::PublicKeys, SBTCResult};
+use bitcoin::{PublicKey, Transaction as BitcoinTransaction};
 use wsts::{bip340::SchnorrProof, common::Signature};
 
-use super::StacksTransaction;
-
-#[derive(Default, Clone, Debug)]
-/// Signers' public keys required for weighted distributed signing
-pub struct PublicKeys {
-    /// Signer ids to public key mapping
-    pub signer_ids: HashMap<u32, ecdsa::PublicKey>,
-    /// Vote ids to public key mapping
-    pub vote_ids: HashMap<u32, ecdsa::PublicKey>,
-}
-
 /// TODO: Define the Message types for DKG round
 /// https://github.com/stacks-network/sbtc/issues/42
 
 /// TODO: Define the Message types for Tx Signning Round
 /// https://github.com/stacks-network/sbtc/issues/43
 
-/// An sBTC transaction needing to be processed by the coordinator
-/// TODO: replace with the core library's SBTCTransaction
-/// This could be a BTC transaction or a STX transaction
-/// depending on https://github.com/Trust-Machines/stacks-sbtc/pull/595
-pub enum SBTCTransaction {
-    /// A commit Bitcoin transaction
-    Commit(TaprootSpendInfo, BitcoinTransaction),
-    /// A withdrawal Stacks transaction
-    Withdawal(StacksTransaction),
-}
-
-/// Revealer trait for revealing BTC commit transactions
-pub trait Reveal {
-    /// Retrieve Commit transactions from Revealer service
-    fn commit_transactions(&self) -> SBTCResult<Vec<(TaprootSpendInfo, BitcoinTransaction)>>;
-    /// Validate the given commit transaction
-    fn validate_commit_transaction(
```

Comment:
Any validation logic should occur within the validate functions in the signer itself, not the coordinator or blockchain observer.